src/components/tasks/card/index.tsx

```diff
@@ -354,6 +425,88 @@ const Card: FC<CardProps> = ({
         };
     };
 
+    const onCancelEditRoute = () => {
+        setIsEditMode(false);
+    };
+
+    const CancelEditButton = () => (
+        <div className={classNames.cancelEditButton}>
+            <Image
+                src="/cancel.png"
+                alt="cancel icon to close edit mode"
+                width={20}
+                height={20}
+                onClick={onCancelEditRoute}
+                tabIndex={0}
+                data-testid="cancel-edit-button"
+            />
+        </div>
+    );
+
+    const handleTitleChange = (value: string) => {
+        setEditedTaskDetails((prev) => ({
+            ...prev,
+            savingText: PENDING,
+        }));
+
+        const response = updateTask({
+            id: cardDetails.id,
+            task: {
+                title: value,
+            },
+        });
+
+        response
+            .unwrap()
+            .then((result) => {
+                setEditedTaskDetails((prev) => ({
+                    ...prev,
+                    savingText: SAVED,
+                }));
+            })
+            .catch((err) => {
+                toast(ERROR, err.data.message);
+            })
+            .finally(() => {
+                setTimeout(() => {
+                    setEditedTaskDetails((prev) => ({
+                        ...prev,
+                        savingText: '',
+                    }));
+                }, 3000);
+            });
+    };
+
+    const debounceTitle = (func: (value: string) => void, delay: number) => {
```

Comment:
in the clearTimeout call. The clearTimeout call is used to cancel a timeout that has already been scheduled. However, the timerIdRef.current variable may be undefined if the debounceTitle function has not been called yet. In this case, the clearTimeout call will throw an error.

Reply:
clearTimeout will only be called if timerId.current has some value, if it’s undefined, it won’t be called. I have already added if condition to check this “if (timerIdRef.current)”. Let me know if I am understanding it correctly.